inst/templates/aggregate_format_data.R

```diff
@@ -0,0 +1,111 @@
+# Meta data ----
+# A template analysis file from erieacoustics
+# Version 0.0.1
+# September 27, 2022
+# M. DuFour
+
+# What it does ----
+# This script will import, combine, and format summarized integration and target strength
+# data from hydroacoustic survey transects. Three .csv files with data associated by
+# BASIN, STRATUM, GRID, LAYER (epilimnion vs. hypolimnion region), and interval are exported.
+#
+# 'hacdat.csv' includes  integration, mean target strength, and fish density estimates.
+#
+# 'histo.csv' includes single target counts for 0.5 dB target strength bins ranging
+# from -64.5 to -20.5 dB.
+#
+# histohac.csv' merges 'hacdat' and 'histo' together.
+
+# How to use ----
+# 1. Ensure that data has imported to Echoview template using '1_import_data_to_template.R'.
+# 2. Ensure that Echoview files have been processed and scrutinized
+# 3. Ensure that transect level data has been exported using '2_export_data_from_EV'
+# 4. Run script below
+
+# ----
+library(readr)
+library(dplyr)
+library(tidyr)
+
+## Gather a list of all files from /3_Ping_Data directory
+allfiles<-dir('3_Ping_Data', recursive = T, full.names = T)
+
+## Import and bind together integration data
+intg<-grep(allfiles, pattern = "intg\\.csv$", value = T)
+intg<-bind_rows(lapply(intg, read_csv))
+intg<-intg %>% filter(!(is.na(Region_name)))
+
+## Import and bind together target strength data
+ts<-grep(allfiles, pattern = "ts\\.csv$", value = T)
+ts<-bind_rows(lapply(ts, read_csv))
+
+## Merge/join integration and target strength data
+hacdat <- left_join(intg, ts, by=c('Region_name', 'Interval', 'Date_M', 'Time_M'))
+
+## remove intervals less than 375 m - typically shorts segment collected beyond specified 5 km transect
+hacdat$IntDist <- hacdat$Dist_E.x-hacdat$Dist_S.x
+hacdat <- filter(hacdat, IntDist >= 375)
+
+## Split Region_name into distinct basin, STRATUM, GRID, and LAYER for hacdat
+hacdat$BASIN <- sapply(strsplit(hacdat$Region_name, "_"), '[', 1)
+hacdat$STRATUM <- sapply(strsplit(hacdat$Region_name, "_"), '[',2)
+hacdat$GRID <- sapply(strsplit(hacdat$Region_name, "_"), '[',3)
+hacdat$LAYER <- sapply(strsplit(hacdat$Region_name, "_"), '[',4)
+
+## replace missing TS values with average TS from INTERVALS within same TRANSECT and LAYER
+hacdat$sigma_bs <- 10^(hacdat$TS_mean/10) # create sigma_bs
+hacdat$sigma_bs <- ifelse(hacdat$TS_mean == 9999, NA,  hacdat$sigma_bs) # assign NAs
+## impute missing sigma_bs_adj data - replace missing and Nv flagged sigma_bs values with mean of interval
+impute.mean <- function(x) replace(x, is.na(x), mean(x, na.rm = TRUE))
+hacdat <- plyr::ddply(hacdat, ~ STRATUM + GRID + LAYER, transform, sigma_bs = impute.mean(sigma_bs))
```

Comment:
@mrdufour Line 60 in aggregate_format_data.R uses a plyr::ddply. Is there a way we can implement this without using plyr? Is there a dplyr alternative? Loading both plyr and dplyr causes namespace conflicts (maybe for select?) and it adds another package dependency.

Reply:
@mrdufour can you test this code chunk in place of lines 59 and 60. I think it is doing the same thing as your code but I'm still getting NA values after I impute but I think it is a problem with my exports, or by problem - I might mean even with imputing I still have NA values and may actually need to impute at the STRATUM + LAYER to get a mean_sigma_bs.
```
hacdat %>% filter(is.na(sigma_bs))
sigma_bs_means <- hacdat %>% group_by(STRATUM, GRID, LAYER) %>% 
  filter(!is.na(sigma_bs)) %>% 
  summarize(mean_sigma_bs = mean(sigma_bs))

hacdat <- left_join(hacdat, sigma_bs_means, by=c('STRATUM', 'GRID', 'LAYER'))
hacdat <- hacdat %>% mutate(ifelse(is.na(sigma_bs), mean_sigma_bs, sigma_bs))
```

Reply:
@HoldenJe Yep, I'll test this afternoon. Also, I'll address the 'gather' function from 'plyr' as well.  Running out to finish up a trawl survey this morning.